File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        mp = {}
        for c in arr:
            mp[c] = mp.get(c, 1) + 1
        return len(set([mp[c] for c in mp])) == len(mp)
        

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.kt
class Solution {
    fun uniqueOccurrences(arr: IntArray): Boolean {
        val map = HashMap<Int, Int>()
        for (i in 0..(arr.size-1)) {
            map.put(arr[i], map.getOrDefault(arr[i], 0) + 1)
        }
        val set = mutableSetOf<Int>()
        for (v in map.values) {
            if (set.contains(v)) return false
            set.add(v)
        }
        return true
    }
}